addons/mail/static/src/xml/chatter.xml

```diff
@@ -112,7 +112,7 @@
         @param {integer} [count]
     -->
     <t t-name="mail.chatter.Attachment.Button">
-        <button class="btn btn-link o_chatter_button_attachment" title="View all the attachments of the current record">
+        <button class="btn btn-link o_chatter_button_attachment" title="View all the attachments of the current record" t-attf-aria-label="Attachments #{count}">
```

Comment:
Done.

> but besides that, is "Attachments 8 <span>8</span>" more helpful than "View all the attachments of the current record <span>8</span>"?

Yes, because "View all" is a button description, not the button itself. Besides, the title is verbalized after all other information: aria-label, button role, etc; so it's still here but with a lower priority.